### PR TITLE
Update K8s dependencies to v0.18.8

### LINKED ISSUE #2489: Update K8s dependencies to 1.18.x
**How to categorize this issue?**

/area robustness
/kind enhancement
/priority normal

**What would you like to be added**:
Update the `vendor` sources in g/g to K8s 1.18.x.

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area open-source
/kind enhancement
/priority normal

**What this PR does / why we need it**:
This PR updates the following dependencies:

- `k8s.io/* v0.18.8`
- `sigs.k8s.io/controller-runtime v0.6.2`
-  `github.com/gardener/external-dns-management v0.7.18`

**Which issue(s) this PR fixes**:
Fixes #2489

**Release note**:

```developer noteworthy
Several Kubernetes dependencies have been updated to `v0.18.8`.
```

### CODE REVIEW

File: pkg/apis/core/v1beta1/conversions.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 	}
 
 	// Add non-generated conversion functions
```
Comment (timebertt) [inline pkg/apis/core/v1beta1/conversions.go]:
```suggestion
```

File: extensions/test/e2e/framework/executor/executor.go [hunk 2]
```diff
@@ -76,15 +76,14 @@ func (e *defaultExecutor) ExecCommandInContainerWithFullOutput(ctx context.Conte
...
-func execWithOptions(ctx context.Context, options execOptions) (stdout, stderr string, err error) {
+func execWithOptions(options execOptions) (stdout, stderr string, err error) {
 	const tty = false
 	req := options.client.Kubernetes().CoreV1().RESTClient().Post().
 		Resource("pods").
 		Name(options.podName).
 		Namespace(options.namespace).
 		SubResource("exec").
-		Param("container", options.containerName).
-		Context(ctx)
```
Comment (rfranzke) [inline extensions/test/e2e/framework/executor/executor.go:87]: Why is that removed?

File: pkg/client/kubernetes/client.go [hunk 3]
```diff
@@ -94,7 +93,7 @@ func NewClientFromBytes(kubeconfig []byte, fns ...ConfigFunc) (Interface, error)
 // be used.
 func NewClientFromSecret(ctx context.Context, c client.Client, namespace, secretName string, fns ...ConfigFunc) (Interface, error) {
 	secret := &corev1.Secret{}
-	if err := c.Get(ctx, kutil.Key(namespace, secretName), secret); err != nil {
+	if err := c.Get(ctx, client.ObjectKey{Namespace: namespace, Name: secretName}, secret); err != nil {
```
Comment (rfranzke) [inline pkg/client/kubernetes/client.go:96]: Why was it changed?

**Review by rfranzke — COMMENTED** [review]
/hold until v1.10 was released, cc @vpnachev

**Review by timebertt — COMMENTED** [review]
Thanks a lot for your work on this!

File: extensions/test/e2e/framework/executor/executor.go (hunk 2 above)
Comment (timuthy) [reply extensions/test/e2e/framework/executor/executor.go:87]: Because there is no `Context(ctx)` any more and btw it never had an affect because we only invoked the `.URL()` function of `req`.

File: pkg/client/kubernetes/client.go (hunk 3 above)
Comment (timuthy) [reply pkg/client/kubernetes/client.go:96]: It was necessary to resolve cyclic dependencies due to the introduction of `DefaultGetOptions`, `DefaultUpdateOptions`, etc. (https://github.com/gardener/gardener/blob/2fb3b07c5c741b432889ce155a89a82cd88532b1/pkg/client/kubernetes/types.go#L70-L76)

File: extensions/test/e2e/framework/executor/executor.go (hunk 2 above)
Comment (timuthy) [reply extensions/test/e2e/framework/executor/executor.go:87]: Basically, removing the context from the request was part of the K8s v0.18.x work in favor of the "executing" functions like `Post`, `GET`, etc. accepting a context.

Comment (rfranzke) [reply extensions/test/e2e/framework/executor/executor.go:87]: Hm ok, but the `Post()` function above doesn't accept a `context`, so I'm confused a bit. Anyways, thanks for the explanation.

Comment (timuthy) [reply extensions/test/e2e/framework/executor/executor.go:87]: Sorry for the confusion. The function which executes the request is `Do()`

https://github.com/kubernetes/client-go/blob/57f65489c11a4d4a978c93ab369c38f53156888a/rest/request.go#L952

Comment (rfranzke) [reply extensions/test/e2e/framework/executor/executor.go:87]: OK, thanks 👍

File: pkg/controllermanager/controller/project/project_control.go [hunk 4]
```diff
@@ -138,11 +139,11 @@ func (c *defaultControl) ReconcileProject(obj *gardencorev1beta1.Project) (bool,
 		return c.delete(context.TODO(), project)
```
Comment (rfranzke) [inline pkg/controllermanager/controller/project/project_control.go]:
```suggestion
		return c.delete(ctx, project)
```

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/controllermanager/controller/project/project_control.go (hunk 4 above)
Comment (timuthy) [reply pkg/controllermanager/controller/project/project_control.go]: Nice catch!

**Comment by timuthy** [comment]
Thanks for your reviews @rfranzke and @timebertt. I've addressed your comments.

**Review by rfranzke — APPROVED** [review]

**Review by timebertt — APPROVED** [review]
Thanks!
/lgtm

**Comment by vpnachev** [comment]
> /hold until v1.10 was released, cc @vpnachev

/unhold
